bool achieved_goal
---
bool succeed
